pkg/controllers/tlspolicy/tlspolicy_controller.go

```diff
@@ -151,6 +152,25 @@ func (r *TLSPolicyReconciler) reconcileResources(ctx context.Context, tlsPolicy
 		return err
 	}
 
+	// validate that the issuer specified exists
+	issuerName := tlsPolicy.Spec.IssuerRef.Name
+	issuerKind := tlsPolicy.Spec.IssuerRef.Kind
+	var issuer client.Object
+	issuerNamespace := ""
+	if issuerKind == v1.ClusterIssuerKind {
+		issuer = &v1.ClusterIssuer{}
+	} else if issuerKind == v1.IssuerKind || issuerKind == "" {
+		issuer = &v1.Issuer{}
+		issuerNamespace = tlsPolicy.Namespace
+	} else {
+		return fmt.Errorf("issuer kind not supported: %s. Must be either ClusterIssuer or Issuer", issuerKind)
+	}
+
+	err = r.Client().Get(ctx, client.ObjectKey{Name: issuerName, Namespace: issuerNamespace}, issuer)
+	if err != nil {
+		return err
+	}
```

Comment:
Looks fine, you might want consider moving into it's own method, presumably we might eventually want to call the same thing from a webhook re. https://github.com/Kuadrant/multicluster-gateway-controller/issues/595

You also might want to double check what the cert-manager code looks like where it loads the Issuer,  it must have similar logic around the Kind etc..

Reply:
The logic is here for cert-manager https://github.com/cert-manager/cert-manager/blob/master/pkg/issuer/helper.go#L57-L74 
I can refactor what we have to be closer. There's a bit in setting up the helper in order to use that function itself but can take a look if you think it's of value?

Reply:
Pushed up a refactor for now.

Reply:
Hmm, looking at that has made me realise that this check isn't going to work unless we have permission to get Issuers and ClusterIssuers. We can obviously add that [see](https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/pkg/controllers/gateway/gateway_controller.go#L81-L87), but all we are really testing is that the mgc controller can get them, not that cert manager can. Presumably we end up in the same situation if cert manager was configured incorrectly for whatever reason.

In an ideal world, we wouldn't need to do any of this and CertManager itself would add a status to the Certificates when the issuer isn't found/accessible and we aggregate the status back from all the Certificates into the policy status :thinking: